internal/controller/oauth/client_credentials_token_provider_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package oauth
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	egv1a1 "github.com/envoyproxy/gateway/api/v1alpha1"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/oauth2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+// MockClientCredentialsTokenSource implements the standard OAuth2 client credentials flow
+type MockClientCredentialsTokenSource struct{}
+
+// FetchToken gets the client secret from the secret reference and fetches the token from provider token URL.
+func (m *MockClientCredentialsTokenSource) Token() (*oauth2.Token, error) {
+	return &oauth2.Token{
+		AccessToken: "token",
+		ExpiresIn:   3600,
+	}, nil
+}
```

Comment:
Since we are able to use the httptest for the mock, we no longer need this `MockClientCredentialsTokenSource`